### PR TITLE
JSON API advice, and warnings cleanup

### PR DESCRIPTION
**Pull request by rustyrussell**
Changelog-None

### CODE REVIEW

File: doc/STYLE.md [hunk 1]
```diff
@@ -177,6 +177,38 @@ fixes as you see other things, and a minimal amount is unavoidable, but
...
+
+All `warning` fields should have unique names which start with
+`warning_`, the value of which should be an explanation.  This allows
+for programs to deal with them sanely, and also perform translations.
+
+## Changing JSON APIs
+
+All JSON API changes need a Changelog line (see below).
+
+You can always add a new JSON field (Changelog-Added), but you cannot
```
Comment (ZmnSCPxj) [inline doc/STYLE.md:201]: Better to clearly say here that you are referring to the return value, not to parameters.

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/3831/commits/635ed6a228bdbf3687005eaa525d28250dfc0e0f

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/3831/commits/13f314b82ff27bfffc6478bb710b509f9a473bf1

File: doc/STYLE.md (hunk 1 above)
Comment (rustyrussell) [reply doc/STYLE.md:201]: This was committed early, so I opened a new PR #3834

**Pull request merged**